### DISCUSSION TITLE
How to translate query with array to MikroORM

### DISCUSSION THREAD

**Discussion by sidzan** [mikro-orm/mikro-orm#2258] [Q&A]
I am having a bit of trouble trying to convert a query from PSQL to mikro-orm. I could not find a proper way to do this, so would seek some advice.

I have 2 queries i want to write in mikro-orm

```
select * from pilots where service_areas @> '[{"province":"BANGKOK"}]';

select * from pilots where service_areas @> '[{"province":"BANGKOK"}]' and service_areas @> '[{"districts":["HUAI_KHWANG"]}]' ;
```

I am not able to translate the simplest one as of now  i.e 
`select * from pilots where service_areas @> '[{"province":"BANGKOK"}]';
`

I have tried the following 
```
await mikroOrm.em.findAndCount(Pilot, {
      serviceAreas: {
        $contains: {
          province: PROVINCES.BANGKOK,
        },
      },
    });
```

which mentions in the error that the `$contain` needs to be outside, but i am not able to grasp the exact syntax.

Any suggestion is much appreciated.

**Comment by B4nan** [comment]
> which mentions in the error that the $contain needs to be outside, but i am not able to grasp the exact syntax.

So what is the error you are talking about? I guess just something from TS, right? Which you can always get around via type assertions (e.g. cast to any or use `expr` helper). The type currently requires array of strings in the value, which might not be 100% correct. 

Btw are you sure the queries you are trying to achieve are correct? Because postgres arrays are using `{}`, not `[]`. When I try to execute such query, it fails on postgres level with malformed array literal.

**Reply by sidzan** [reply]
Yes, the queries i am trying to run are working fine with psql, more info at [here](https://stackoverflow.com/a/28497341)
```
select * from pilots where service_areas @> '[{"province":"BANGKOK"}]' and service_areas @> '[{"districts":["HUAI_KHWANG"]}]' ;
```


This is my Entity

```
@Embedded(() => ServiceArea, { array: true })
readonly serviceAreas: ServiceArea[];
  
  
@Embeddable()
export class ServiceArea {
  @Property({ type: String })
  @Enum(() => PROVINCES)
  province!: PROVINCES;

  @Property({ type: String })
  @Enum(() => DISTRICTS)
  districts!: DISTRICTS[];
}
```
for now i managed to achieve the query i wanted with the following
```
    const [item, count] = await mikroOrm.em.findAndCount(Pilot, {
      $and: [
        {
          serviceAreas: {
            $contains: `[{"province":"${PROVINCES.BANGKOK}"}]`,
          },
        },
        {
          serviceAreas: {
            $contains: `[{"districts":["${DISTRICTS.HUAI_KHWANG}"]}]`,
          },
        },
      ],
    });

```

But there is a lot of interpolation going on and may be there is a better way to write this query?
